x/emissions/module/abci.go

```diff
@@ -13,16 +14,28 @@ import (
 )
 
 func EndBlocker(ctx context.Context, am AppModule) error {
-	defer telemetry.ModuleMeasureSince(emissionstypes.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
-
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	blockHeight := sdkCtx.BlockHeight()
 	sdkCtx.Logger().Debug("---------------- Emissions EndBlock -------------------", "blockHeight", blockHeight)
+
 	moduleParams, err := am.keeper.GetParams(sdkCtx)
 	if err != nil {
 		sdkCtx.Logger().Error("Error Getting module params", err)
 		return err
 	}
+
+	defer func() {
+		telemetry.ModuleMeasureSince(emissionstypes.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
```

Comment:
The `time.Now` in this line must be executed very first which here is called in the deferred func, and the `telemetry.ModuleMeasureSince` must be executed very last which is not the case either.

As defer are LIFOed we should keep this in a first defer:
```
defer telemetry.ModuleMeasureSince(emissionstypes.ModuleName, telemetry.Now(), telemetry.MetricKeyEndBlocker)
```